tgui/packages/tgui/interfaces/RequestsConsole/MessageViewTab.tsx

```diff
@@ -0,0 +1,80 @@
+import { useBackend } from '../../backend';
+import { BlockQuote, Button, LabeledList, NoticeBox, Section, Stack } from '../../components';
+import { decodeHtmlEntities } from 'common/string';
+import { RequestPriority, RequestsData, RequestType } from './Types';
+
+export const MessageViewTab = (props, context) => {
+  const { act, data } = useBackend<RequestsData>(context);
+  const { messages = [] } = data;
+  return (
+    <Section fill scrollable>
+      {!!messages.length && (
+        <Stack vertical>
+          {messages.map((message) => (
+            <MessageDisplay key={message.received_time} message={message} />
+          ))}
+        </Stack>
+      )}
+    </Section>
+  );
+};
+
+export const MessageDisplay = (props, context) => {
+  const { act } = useBackend(context);
+  const { message } = props;
```

Comment:
is message type safe?

Reply:
I have defined the props now, should be strictly typed.